constraint/pkg/client/client.go

```diff
@@ -492,27 +482,14 @@ func (c *Client) AddConstraint(ctx context.Context, constraint *unstructured.Uns
 	if err := c.validateConstraint(constraint, false); err != nil {
 		return resp, err
 	}
+	if err := c.backend.driver.AddConstraint(ctx, constraint); err != nil {
+		return resp, err
+	}
 	for _, target := range entry.Targets {
-		relPath, err := createConstraintPath(target, constraint)
-		// If we ever create multi-target constraints we will need to handle this more cleverly.
-		// the short-circuiting question, cleanup, etc.
-		if err != nil {
-			errMap[target] = err
-			continue
-		}
-		if err := c.backend.driver.PutData(ctx, relPath, constraint.Object); err != nil {
-			errMap[target] = err
-			continue
-		}
 		resp.Handled[target] = true
 	}
-
-	if len(errMap) == 0 {
-		c.constraints[constraint.GroupVersionKind().GroupKind()][subPath] = constraint.DeepCopy()
-		return resp, nil
-	}
-
-	return resp, &errMap
+	c.constraints[constraint.GroupVersionKind().GroupKind()][subPath] = constraint.DeepCopy()
```

Comment:
Make sure to guard this write with `c.mtx` so concurrent calls don't fight. Ditto other calls to c.constraints.

Reply:
Also - I'm pretty sure `c.constraints` isn't needed for anything any more. Can you add a TODO to the field declaration, and add an issue to remove the `constraints` field from `Client`?

Reply:
Add TODO with link to #187, it would break `client.GetConstraint` if it has a usage.
The updates for c.constraints is guarded by c.mtx at the beginning of AddConstraint